src/types/ant.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { z } from 'zod';
+
+import { Logger } from '../common/logger.js';
+import { ARWEAVE_TX_REGEX } from '../constants.js';
+import { AoMessageResult, WalletAddress, WriteOptions } from './common.js';
+
+/**
+ * example error:
+ *  {
+        "code": "custom",
+        "message": "Must be an Arweave Transaction ID",
+        "path": [
+          "Records",
+          "record1",
+          "transactionId"
+        ]
+      },
+ */
+export const ArweaveTxIdSchema = z
```

Comment:
We dont current support eth or solana address in our ants, if/when we do this can be made into a union